### ISSUE TITLE
ActionController::Parameters should implement `reverse_merge`

### ISSUE THREAD

**Issue by derekprior** [rails/rails#28353]
#`reverse_merge` is useful for setting defaults in hashes (and also controller parameters). `Parameters` already implements `merge`, so it should likely implement `reverse_merge` as well.

### Steps to reproduce

```ruby
params = ActionController::Parameters.new({ foo: :bar })
defaults = { baz: :bang }
params.reverse_merge(defaults)
```

### Expected behavior

No deprecations

### Actual behavior

> DEPRECATION WARNING: Method reverse_merge is deprecated and will be removed in Rails 5.1, as `ActionController::Parameters` no longer inherits from hash. Using this deprecated behavior exposes potential security problems. If you continue to use this method you may be creating a security vulnerability in your app that can be exploited. Instead, consider using one of these documented methods which are not deprecated: http://api.rubyonrails.org/v5.0.0.1/classes/ActionController/Parameters.html

### System configuration

**Rails version**: 5.0.0.1
**Ruby version**: 2.3.3

**Comment by sikachu** [comment]
I agree that it's a good addition. Would you mind submitting a PR with implementation and test?

**Comment by pixeltrix** [comment]
It would also make the new `direct` api example easier (see #28301).

**Comment by maclover7** [comment]
Looks like we have [denied PRs](https://github.com/rails/rails/pull/27198#issuecomment-263586779) in the past to add this method...

cc @rafaelfranca

**Comment by derekprior** [comment]
I think we should reconsider. Without `reverse_merge`, it's a bit cumbersome to default parameters in a controller.

```ruby
def post_params
  defaults.merge(params.require(:post).permit(:body))
end

def defaults
  ActionController::Parameters.new({user: current_user})
end
```

Note that only works on Rails 5.1 (there is a deprecation when you do this in Rails 5.0, which I believe has been fixed). In rails 5.0 you would have to do this (or use `[]` to mutate more directly).

```ruby
def post_params
  params.require(:post).permit(:body).to_h.reverse_merge({user: current_user})
end
```

`reverse_merge` is just really very useful for setting defaults. In fact, if we want to make that explicit (and further divorce `ActionController::Parameters` from `Hash` it could be `with_defaults` instead of `reverse_merge`).

**Comment by rafaelfranca** [comment]
Fixed by #28355
